backend/lib/use-cases/dataflows/sync-cases.ts

```diff
@@ -4,34 +4,38 @@ import Factory, { getCasesGateway } from '../../factory';
 import { getCamsError } from '../../common-errors/error-utilities';
 import { CasesSyncState } from '../gateways.types';
 import { randomUUID } from 'node:crypto';
+import { CamsError } from '../../common-errors/cams-error';
 
 const MODULE_NAME = 'SYNC-CASES-USE-CASE';
 
-async function getCaseIds(context: ApplicationContext, lastRunTxId?: string) {
+async function getCaseIds(context: ApplicationContext, lastSyncDate?: string) {
   try {
-    const runtimeStateRepo = Factory.getCasesSyncStateRepo(context);
+    const now = new Date().toISOString();
 
     let syncState: CasesSyncState;
-    if (lastRunTxId) {
+    if (lastSyncDate) {
       syncState = {
         id: randomUUID(),
         documentType: 'CASES_SYNC_STATE',
-        txId: lastRunTxId,
+        lastSyncDate,
       };
     } else {
+      const runtimeStateRepo = Factory.getCasesSyncStateRepo(context);
       syncState = await runtimeStateRepo.read('CASES_SYNC_STATE');
     }
 
     const casesGateway = getCasesGateway(context);
-    const { caseIds, lastTxId } = await casesGateway.getCaseIdsAndMaxTxIdToSync(
-      context,
-      syncState.txId,
-    );
+    const start = syncState.lastSyncDate;
+    if (!start) {
+      throw new CamsError(MODULE_NAME);
```

Comment:
We should add a description of the error here.
```suggestion
      throw new CamsError(MODULE_NAME, {message: 'Last sync date not available.'});
```